### DISCUSSION TITLE
Model Predicate Provider Registry

### DISCUSSION THREAD

**Discussion by JCS-Mecabricks** [FabricMC/fabric-api#4295] [Mod Dev Support]
### Description

I am updating my mod to 1.21.4 but I use a Model Predicate Provider Registry for my bow. How do I do it now?

**Comment by Linguardium** [comment]
Use data generation
in your item model generator, set the model to use the new model properties. see ItemModels class and its usages

**Comment by JCS-Mecabricks** [comment]
My item is a bow.

**Reply by Linguardium** [reply]
ok. do the thing i said to do

**Comment by JCS-Mecabricks** [comment]
Thanks. I will try.

**Comment by JCS-Mecabricks** [comment]
I was able to do it but the animation for the bow pull doesn't show. Also the arrow isn't showing. I would appreciate if you could help me.
